src/fsharp/CompileOps.fs

```diff
@@ -3832,12 +3832,12 @@ and [<Sealed>] TcImports(tcConfigP: TcConfigProvider, initialResolutions: TcAsse
     let mutable dllTable: NameMap<ImportedBinary> = NameMap.empty
     let mutable ccuInfos: ImportedAssembly list = []
     let mutable ccuTable: NameMap<ImportedAssembly> = NameMap.empty
-    let mutable disposeActions = []
+    let disposeActions = ResizeArray()
```

Comment:
This remind me of something I was looking at:

https://github.com/dotnet/fsharp/blob/0a0889e3564d0ff745119cca2371e9ea78a82a99/src/fsharp/ConstraintSolver.fs#L370-L376

this is used a lot in `ResolveOverloading` and is probably short lived so not the same kind of usage pattern.

I was wondering about the impact of all allocations happening in `ResolveOverloading` as many things are based on exception objects, I guess just as a convenience to make the error message, but this is only useful when failing to resolve an overload which is a corner case rather than the default.

This happens while checking each of the candidates, and there are several passes (using trace objects), since Exception is a very large object, generally larger itself than the few references we end up needing once the resolution failed to assemble our own exceptions, I intuit that lots of allocations could be saved there, but I also intuit that the refactoring of the data structures wouldn't be a low hanging fruit.

In any case, exciting to see the work you are doing dealing with allocations in the compiler 👍.

Reply:
I would wonder too. I've seen this kind of pattern as well in other places. It depends on what the use case is. You can think of this pattern like simply swapping state rather than changing it underneath (which is what ResizeArray) does.

For this PR in particular, using the `list` state swapping pattern is fine but I need a reference to the actual state without capturing `TcImports` inside the object doing the disposal. I would have to make this a ref cell which I didn't want, and chose ResizeArray instead.